From b1cdf7c5d4df3221504c34010985254f4261b3dd Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Fri, 24 Jan 2014 11:55:27 +0100
Subject: [PATCH] Revert "dmraid: let dmraid setup the partitions"

This reverts commit fbf717086e8b02947a27f55d1759cccd1cb89e99.

dmraid seems to use "p" as a seperator by default.

Reverting to kpartx, until this is fixed.
---
 modules.d/90dmraid/dmraid.sh | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/modules.d/90dmraid/dmraid.sh b/modules.d/90dmraid/dmraid.sh
index dfd0f1cb..3753ddd7 100755
--- a/modules.d/90dmraid/dmraid.sh
+++ b/modules.d/90dmraid/dmraid.sh
@@ -27,7 +27,8 @@ if [ -n "$DM_RAIDS" ] || getargbool 0 rd.auto; then
             for s in $SETS; do
                 if [ "${s##$r}" != "$s" ]; then
                     info "Activating $s"
-                    dmraid -ay -i --rm_partitions "$s" 2>&1 | vinfo
+                    dmraid -ay -i -p --rm_partitions "$s" 2>&1 | vinfo
+                    [ -e "/dev/mapper/$s" ] && kpartx -a "/dev/mapper/$s" 2>&1 | vinfo
                     udevsettle
                 fi
             done
@@ -36,7 +37,8 @@ if [ -n "$DM_RAIDS" ] || getargbool 0 rd.auto; then
     # scan and activate all DM RAIDS
         for s in $SETS; do
             info "Activating $s"
-            dmraid -ay -i --rm_partitions "$s" 2>&1 | vinfo
+            dmraid -ay -i -p --rm_partitions "$s" 2>&1 | vinfo
+            [ -e "/dev/mapper/$s" ] && kpartx -a "/dev/mapper/$s" 2>&1 | vinfo
         done
     fi
 
